bin/worker

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env ruby
+
+require_relative '../config/bootstrap.rb'
+require 'routemaster/services/thread_group'
+require 'routemaster/services/worker'
+require 'routemaster/services/ticker'
+require 'routemaster/mixins/log'
+
+include Routemaster::Mixins::Log
+
+Routemaster.configure(
+  # One connection for each worker thread, one for the aux worker, and one for
+  # the tickers to enqueue their jobs
+  redis_pool_size: Integer(ENV.fetch('ROUTEMASTER_WORKER_THREADS')) + 2
+)
+
+_log.info { 'creating thread group' }
+Routemaster::Services::ThreadGroup.new.tap do |group|
+  # ticker threads
+  group.add Routemaster::Services::Ticker.new(
+    queue: Routemaster.aux_queue, name: 'schedule', every: 100, delay: false
+  ), name: 'ticker.schedule'
+  group.add Routemaster::Services::Ticker.new(
+    queue: Routemaster.aux_queue, name: 'autodrop', every: 10_000
+  ), name: 'ticker.autodrop'
+  group.add Routemaster::Services::Ticker.new(
+    queue: Routemaster.aux_queue, name: 'monitor', every: 10_000
+  ), name: 'ticker.monitor'
+
+  # aux worker thread
+  group.add Routemaster::Services::Worker.new(
+    queue: Routemaster.aux_queue
+  ), name: "worker.aux"
+
+  # main worker threads
+  Integer(ENV.fetch('ROUTEMASTER_WORKER_THREADS')).times do |idx|
+    group.add Routemaster::Services::Worker.new(
+      queue: Routemaster.batch_queue
+    ), name: "worker.main.#{idx + 1}"
+  end
+
+  _log.info { 'trapping signals for clean exit' }
+  %w(INT TERM QUIT).each do |signal|
+    Kernel.trap(signal) { Thread.new { group.stop } }
```

Comment:
It is: the method is idempotent. Like `Routemaster::Services::Thread#stop` above, it just signals the individual threads to stop and closes the error queue (itself idempotent), which stops the error watcher thread.